merlin/table/tensor_column.py

```diff
@@ -108,6 +130,16 @@ def __eq__(self, other):
             and self.device == other.device
         )
 
+    def _construct_shape(self, values, offsets):
+        if offsets is not None:
+            num_rows = len(offsets) - 1
+            row_lengths = offsets[1:] - offsets[:-1]
+            num_cols = int(row_lengths[0]) if all(row_lengths == row_lengths[0]) else None
+            shape = Shape((num_rows, num_cols))
```

Comment:
would we like to support 2-dimensional `values` here (e.g. corresponding to arrays in a 3-d ragged column). For example a ragged list column containing lists of fixed-size embeddings.

for example a column like (shape: (2, None, 2) ):

```
    [[1, 2], [3, 4]]
            [[3, 5]]
```

with `values` (shape: (3, 2) ):

```
[
  [1, 2],
  [3, 4],
  [3, 5]
]
```

and `offsets` (shape: (3,) ):

```
[0, 2, 3]
```

Could be an enhancement for later if we don't have a reqirement for this now

Reply:
We would like to support those, which @jperez999 is starting to take a look at. I like this thought about how to encode the values in a fixed shape tensor—let's try it out and see if it works for our use cases. 👍🏻 

I think I'd opt to leave it for a future PR though, since we only need support for 2-d shapes in Systems (which is the current `TensorTable` test bed) for now.